projects/core/src/occ/adapters/product/converters/occ-product-search-page-normalizer.ts

```diff
@@ -20,11 +20,40 @@ export class OccProductSearchPageNormalizer
       ...target,
       ...(source as any),
     };
+    this.normalizeUselessFacets(target);
     if (source.products) {
       target.products = source.products.map(product =>
         this.converterService.convert(product, PRODUCT_NORMALIZER)
       );
     }
     return target;
   }
+
+  /**
+   * The (current) backend returns facets with values that do not contribute
+   * to the facet navigation much, as the number in the result list will not get
+   * affected when they got applied. A ticket has been created to fix this
+   * behaviour, see https://jira.hybris.com/browse/CS-427.
+   *
+   * As long as this is not in place, we manually filter the facet from the list;
+   * any facet that does not have a count < the total results will be dropped from
+   * the facets.
+   */
+  private normalizeUselessFacets(target: ProductSearchPage): void {
+    target.facets = [].concat(
```

Comment:
Not sure what's the idea behing `[].concat`? Maybe `target.facets = target.facets.filter(...)` is enough?